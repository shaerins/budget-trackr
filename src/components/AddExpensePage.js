import { connect } from "react-redux";
import { startAddExpense } from "../actions/expenses";
import React from "react";
import ExpenseForm from "./ExpenseForm";

export class AddExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.startAddExpense(expense);
        this.props.history.push("/");
    };
    render() {
        return (
            <>
                <h1>Add Expense</h1>
                <ExpenseForm
                    onSubmit={this.onSubmit} />
            </>
        );
    }
}

const mapDispatchToProps = (dispatch) => ({
    startAddExpense: (expense) => dispatch(startAddExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);