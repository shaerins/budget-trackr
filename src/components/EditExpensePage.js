import React from "react";

const EditExpensePage = () => (
    <div>
        edit expenses page
    </div>
);

export default EditExpensePage;