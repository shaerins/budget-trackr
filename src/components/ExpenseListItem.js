import React from "react";
import { removeExpense } from "../actions/expenses";
import { connect } from "react-redux";

const ExpenseListItem = ({ description, amount, createdAt, id, dispatch }) => (
    <div>
        <h3>{description}</h3>
        <p>Amount: ${amount} - Date: {createdAt}
            <button onClick={() => {
                dispatch(removeExpense({id}))
            }}
            >Remove</button>
        </p>
    </div >
);

export default connect()(ExpenseListItem);