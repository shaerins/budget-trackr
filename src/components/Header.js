import React from "react";
import { NavLink } from "react-router-dom";

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink>
        <NavLink to="/create" activeClassName="is-active" exact={true}>Create</NavLink>
        <NavLink to="/Edit" activeClassName="is-active" exact={true}>Edit</NavLink>
        <NavLink to="/help" activeClassName="is-active" exact={true}>Help</NavLink>
    </header>
);

export default Header;