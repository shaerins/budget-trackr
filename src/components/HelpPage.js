import React from "react";

const HelpPage = () => (
    <>
        help page
    </>
);

export default HelpPage;