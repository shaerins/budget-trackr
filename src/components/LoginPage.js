import React from "react";
import { connect } from "react-redux";
import { startLogin } from "../actions/auth";

export const LoginPage = ({ startLogin }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__header">budget trackr</h1>
            <p>Keep track of your spending.</p>
            <button className="button" onClick={startLogin}>Login with Google</button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);