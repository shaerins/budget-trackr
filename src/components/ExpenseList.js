import React from "react";
import { connect } from "react-redux";
import ExpenseListItem from "./ExpenseListItem";
import selectedExpenses from "../selectors/expenses";

export const ExpenseList = (props) => (
    <div className="content-container">
        <div className="list-header">
            <div className="show-on-mobile">Expenses</div>
            <div className="show-on-desktop">Expense</div>
            <div className="show-on-desktop">Amount</div>
        </div>
        <div className="list-body">
            {
                props.expenses.length === 0 ? (
                    <span className="list-item list-item__message">You have no expenses.</span>
                ) : (
                        props.expenses.map((expense) => {
                            return <ExpenseListItem key={expense.id} {...expense} />
                        })
                    )
            }
        </div>
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectedExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpenseList);