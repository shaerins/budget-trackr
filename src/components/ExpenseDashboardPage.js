import React from "react";
import ExpenseList from "./ExpenseList";
import ExpenseListFilters from "./ExpenseListFilters";
import ExpensesSummary from "./ExpensesSummary";

const ExpenseDashboardPage = () => (
    <React.Fragment>
        <ExpensesSummary />
        <ExpenseListFilters />
        <ExpenseList />
    </React.Fragment>
);

export default ExpenseDashboardPage;