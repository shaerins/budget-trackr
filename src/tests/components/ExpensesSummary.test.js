import React from "react";
import { ExpensesSummary } from "../../components/ExpensesSummary";
import { shallow } from "enzyme";

test("should render expenses summary with one expense", () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={2345} />);
    expect(wrapper).toMatchSnapshot();
});

test("should render expenses summary with one expense", () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={25} expensesTotal={12345634} />);
    expect(wrapper).toMatchSnapshot();
});
