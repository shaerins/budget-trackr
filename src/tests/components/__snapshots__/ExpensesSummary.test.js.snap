// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenses summary with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $23.45
    .
  </h1>
</div>
`;

exports[`should render expenses summary with one expense 2`] = `
<div>
  <h1>
    Viewing 
    25
     
    expenses
     totalling 
    $123,456.34
    .
  </h1>
</div>
`;
